File: server/src/modules/setup/service_operations/update_school_settings.rs
use crate::modules::setup::repository::SetupRepository;
use crate::modules::setup::schema::{SchoolSettingsResponse, UpdateSchoolSettingsRequest};
use crate::utils::AppResult;

pub async fn update_school_settings(
    repo: &SetupRepository,
    request: UpdateSchoolSettingsRequest,
) -> AppResult<SchoolSettingsResponse> {
    let updated = repo
        .update_settings(
            None,
            Some(Some(request.school_name)),
            request.school_region.map(Some),
            request.school_division.map(Some),
            request.school_year.map(Some),
        )
        .await?;

    Ok(SchoolSettingsResponse {
        school_code: updated.school_code,
        school_name: updated.school_name,
        school_region: updated.school_region,
        school_division: updated.school_division,
        school_year: updated.school_year,
    })
}

File: client/lib/domain/setup/usecases/update_school_settings.dart
import 'package:likha/core/sync/mutation_result.dart';
import 'package:likha/core/utils/typedef.dart';
import 'package:likha/domain/setup/entities/school_settings.dart';
import 'package:likha/domain/setup/repositories/setup_repository.dart';

class UpdateSchoolSettings {
  final SetupRepository _repository;

  UpdateSchoolSettings(this._repository);

  ResultFuture<MutationResult<SchoolSettings>> call({
    required String schoolName,
    required String schoolRegion,
    required String schoolDivision,
    required String schoolYear,
    required String schoolCode,
    String? schoolDistrict,
  }) {
    return _repository.updateSchoolSettings(
      schoolName: schoolName,
      schoolRegion: schoolRegion,
      schoolDivision: schoolDivision,
      schoolYear: schoolYear,
      schoolCode: schoolCode,
      schoolDistrict: schoolDistrict,
    );
  }
}
